src/services/music-providers/youtube-music/youtube-music.service.ts

```diff
@@ -0,0 +1,286 @@
+        }
+
+        const track = findTracks.find((track) =>
+            this.checkThatTrackIsSuitable(
+                name,
+                artists,
+                track.title,
+                this.artistsToNameArray(track.artists),
+            ),
+        );
+
+        if (!track) {
+            return null;
+        }
+
+        const result = {
+            id: track.videoId,
+            name: track.title,
+            artists: track.artists.map((a) => a.name),
+            source: track,
+        };
+        this.cache.set(searchQuery, result);
+
+        return result;
+    }
+
+    async addTracksToPlaylist(
+        trackIds: string[],
+        playlist: Playlist,
+    ): Promise<void> {
+        await retry<PlaylistLibmuse>(
+            async () => await add_playlist_items(playlist.id, trackIds),
+            async () =>
+                this.notifyThatFunctionCompletedWithError(
+                    'addTracksToPlaylist',
+                ),
+        );
+    }
+
+    async removeTracksFromPlaylist(
+        tracks: Track[],
+        playlist: Playlist,
+    ): Promise<void> {
+        const playlistFromServer: PlaylistLibmuse = await this.getPlaylist(
+            playlist.id,
+            { limit: this.getPlaylistTracksLimit },
+        );
+
+        const trackWithId = tracks.filter((t) => {
+            if (t?.id) {
+                return true;
+            }
+
+            this.logService.warn(
+                `Track ${t.name} by ${t.artists} was not removed in youtube. Reason id is absent`,
+            );
+        });
+
+        const trackFromServer = playlistFromServer.tracks.filter(
+            (trackFromServer) =>
+                trackWithId.some(
+                    (propsTrack) => propsTrack.id === trackFromServer.videoId,
+                ),
+        );
+
+        const trackFromServerWithSetVideoId = trackFromServer.filter((t) => {
+            if (t.setVideoId) {
+                return true;
+            }
+
+            this.logService.warn(
+                `Track ${t.title} by ${t.artists} was not removed in youtube. Reason setVideoId is absent`,
+            );
+        });
+
+        if (!trackFromServerWithSetVideoId.length) {
+            return;
+        }
+
+        await this.removePlaylistItemsWrapperOverLibmuseMethod(
+            playlist.id,
+            trackFromServerWithSetVideoId.map((t) => ({
+                videoId: t.videoId,
+                setVideoId: t.setVideoId as string,
+            })),
+        );
+    }
+
+    private artistsToNameArray(artists: SongArtist[]) {
+        let result: string[] = [];
+
+        for (const artist of artists) {
+            if (artist.id) {
+                result.push(artist.name);
+                continue;
+            }
+
+            const names = artist.name.replace(/, & | & | and | и /g, ', ');
+            result = [...result, ...names.split(', ')];
+        }
+
+        return result;
+    }
+
+    private checkThatTrackIsSuitable(
```

Comment:
Consider renaming `checkThatTrackIsSuitable` to a more descriptive name like `isTrackMatchingCriteria`. This improves readability and makes it immediately clear what the function is checking.